File: src/java/NeonNumber.java
package java;

public class NeonNumber {

    public static boolean isNeonNumber(long num) {
        long sqrSum = (long) Math.pow(num,2);
        int sum = 0;

        while(sqrSum > 0) {
            sum += sqrSum % 10;
            sqrSum /= 10;
        }

        return num == sum;
    }

}

File: src/C++/NeonNumber.cpp
#include <iostream>
using namespace std;


bool isNeonNumber(long n) {
    long sqr = n * n;
    long sum = 0;
    while (sqr > 0) {
        sum = sum + sqr % 10;
        sqr = sqr / 10;
    }

    return sum == n;
}
